.github/workflows/create_pr_for_changelog.yml

```diff
@@ -0,0 +1,52 @@
+name: Generate changelog PR
+
+on:
+  push:
+    branches: [master]
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Generate the full changelog
+        uses: orhun/git-cliff-action@v4
+        id: git_cliff
+        with:
+          config: cliff.toml
+          args: --verbose --exclude-path "data/" --bump
+        env:
+          OUTPUT: CHANGELOG.md
+
+      - name: Generate latest changes
+        uses: orhun/git-cliff-action@v4
+        id: git_cliff_latest
+        with:
+          config: cliff.toml
+          args: --verbose --latest --strip header --exclude-path "data/" --bump --unreleased
+
+      - name: Create Pull Request
+        if: steps.git_cliff.outputs.version != 'null'
+        uses: peter-evans/create-pull-request@v7
+        with:
+          add-paths: |
+            CHANGELOG.md
+          base: master
+          commit-message: "chore: update CHANGELOG.md for ${{ steps.git_cliff.outputs.version }}"
+          signoff: false
```

Comment:
Would setting this to `true` fail? What would be required to have a signed-off commit here?

Reply:
Good point. I'll add commit signing too.